kaitaistruct.py

```diff
@@ -320,10 +451,7 @@ def read_bytes_term(self, term, include_term, consume_term, eos_error):
     def ensure_fixed_contents(self, expected):
         actual = self._io.read(len(expected))
         if actual != expected:
-            raise Exception(
-                "unexpected fixed contents: got %r, was waiting for %r" %
-                (actual, expected)
-            )
+            raise ValidationNotEqualError(expected, actual, self._io, __file__)
```

Comment:
Please revert this. The `ensure_fixed_contents` method is left here only for backward compatibility with generated code by 0.8 compiler and lower (the 0.9 compiler-generated code no longer uses this method), and thus it makes a little sense why raise a more specific type here. It can actually break the compatibility, because you completely changed the error message by doing this (if some KS user does string matching on the error message to distinguish this exception among others, it probably won't work anymore).

And unsurprisingly, it doesn't have to do anything with serialization again.